src/client/encrypt.js

```diff
@@ -18,7 +18,7 @@ module.exports = function (client, options) {
         client.end()
         return
       }
-      if (options.haveCredentials) {
+      if ((options.auth === 'microsoft' && !options.haveCredentials) || options.haveCredentials) {
```

Comment:
If you set `options.haveCredentials = true` in microsoftAuth.js, you don't need this change, and the PR will be a bit more self-contained.

Reply:
Well rom said he wanted to switch between the auth endpoints using auth and options.haveCredentials is a Mojang auth thing and auth is a Microsoft thing but if you want me to ditch the auth check here I can.

Reply:
`haveCredentials` is not a mojang thing https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/src/client/auth.js#L22
its just a boolean to say if the user provided a password. If you are using a ms account it should have the same definition.